File: C#SrcFiles/Cathulu.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Cathulu : MonoBehaviour
{
    public HealthBarScript healthbar;
    private GameObject player;
    private float chaseSpeed;
    private float Distance;
    private float MaxDistance;
    private float Xpos;
    private float Ypos;
    public int maxHealth;
    public int currHealth;
    void Start()
    {
        player = GameObject.FindGameObjectWithTag("Player");
        MaxDistance = 30f;
        chaseSpeed = 8f;
        maxHealth = currHealth = 100;
        healthbar.SetMaxHealth(maxHealth);
        healthbar.SetHealth(currHealth);
    }

    // Update is called once per frame
    void Update()
    {
        Xpos = transform.position.x;
        Ypos = transform.position.y;
        Distance = Vector2.Distance(transform.position, player.transform.position);
        Vector2 direction = player.transform.position - transform.position;
        direction.Normalize();
        float angle = Mathf.Atan2(direction.y, direction.x) * Mathf.Rad2Deg;

        transform.position = Vector2.MoveTowards(this.transform.position, player.transform.position, chaseSpeed * Time.deltaTime);
        transform.rotation = Quaternion.Euler(Vector3.forward * (angle + 180));


        if (transform.position.y <= 20f) 
        {
            if (player.transform.position.x > transform.position.x) 
            {
                Xpos += chaseSpeed * Time.deltaTime;
            }
            else
            {
                Xpos -= chaseSpeed * Time.deltaTime;
            }
            transform.position = new Vector3(Xpos, transform.position.y, transform.position.z);
        }
        if (transform.position.y <= 19f)
        {
            Ypos += chaseSpeed * Time.deltaTime;
            transform.position = new Vector3(transform.position.x, Ypos, transform.position.z);
        }

    }
    public void TakeDamage(int Damage)
    {
        currHealth -= Damage;
        healthbar.SetHealth(currHealth);
    }
}

File: C++SrcFiles/Cathulu.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class Cathulu {
private:
    GameObject* player;
    float chaseSpeed;
    float distance;
    float maxDistance;
    float xPos;
    float yPos;
    int maxHealth;
    int currHealth;
    HealthBarScript healthbar;

public:
    Cathulu() {
        player = GameObject::FindWithTag("Player");
        maxDistance = 30.0f;
        chaseSpeed = 8.0f;
        maxHealth = currHealth = 100;
        healthbar.SetMaxHealth(maxHealth);
        healthbar.SetHealth(currHealth);
    }

    void Update() {
        xPos = transform->position.x;
        yPos = transform->position.y;
        distance = Vector2::Distance(transform->position, player->transform->position);

        // Calculate movement direction
        Vector2 direction = player->transform->position - transform->position;
        direction.Normalize();
        float angle = atan2(direction.y, direction.x) * Mathf::Rad2Deg;

        // Move towards player
        transform->position = Vector2::MoveTowards(transform->position, player->transform->position, chaseSpeed * Time.deltaTime);
        transform->rotation = Quaternion::Euler(Vector3::forward * (angle + 180.0f));

        // Limit vertical movement
        if (transform->position.y <= 20.0f) {
            if (player->transform->position.x > transform->position.x) {
                xPos += chaseSpeed * Time.deltaTime;
            } else {
                xPos -= chaseSpeed * Time.deltaTime;
            }

            transform->position = new Vector3(xPos, transform->position.y, transform->position.z);
        }

        if (transform->position.y <= 19.0f) {
            yPos += chaseSpeed * Time.deltaTime;
            transform->position = new Vector3(transform->position.x, yPos, transform->position.z);
        }
    }

    void TakeDamage(int damage) {
        currHealth -= damage;
        healthbar.SetHealth(currHealth);
    }
};
